micrometer/src/main/java/org/wildfly/quickstarts/micrometer/RootResource.java

```diff
@@ -0,0 +1,118 @@
+    static final String METER_DUPLICATED_COUNTER = "prime_duplicatedCounter";
+    private long highestPrimeNumberSoFar = 2;
+
+    @Inject
+    private MeterRegistry registry;
+
+    private Counter performCheckCounter;
+    private Counter originalCounter;
+    private Counter duplicatedCounter;
+
+    @PostConstruct
+    private void createMeters() {
+        Gauge.builder(METER_HIGHEST_SO_FAR, () -> highestPrimeNumberSoFar)
+                .description("Highest prime number so far.")
+                .register(registry);
+        performCheckCounter = Counter
+                .builder(METER_PERFORMED_CHECKS)
+                .description("How many prime checks have been performed.")
+                .register(registry);
+        originalCounter = Counter
+                .builder(METER_DUPLICATED_COUNTER)
+                .tags(List.of(Tag.of("type", "original")))
+                .register(registry);
+        duplicatedCounter = Counter
+                .builder(METER_DUPLICATED_COUNTER)
+                .tags(List.of(Tag.of("type", "copy")))
+                .register(registry);
+    }
+
+    @GET
+    @Path("/prime/{number}")
+    public String checkIfPrime(@PathParam("number") long number) throws Exception {
+        performCheckCounter.increment();
+
+        Timer timer = registry.timer("prime.timer");
+
+        return timer.recordCallable(() -> {
+
+            if (number < 1) {
+                return "Only natural numbers can be prime numbers.";
+            }
+
+            if (number == 1) {
+                return "1 is not prime.";
+            }
+
+            if (number == 2) {
+                return "2 is prime.";
+            }
+
+            if (number % 2 == 0) {
+                return number + " is not prime, it is divisible by 2.";
+            }
+
+            for (int i = 3; i < Math.floor(Math.sqrt(number)) + 1; i = i + 2) {
+                try {
+                    Thread.sleep(10);
+                } catch (InterruptedException e) {
+                    //
+                }
+                if (number % i == 0) {
+                    return number + " is not prime, is divisible by " + i + ".";
+                }
+            }
+
+            if (number > highestPrimeNumberSoFar) {
+                highestPrimeNumberSoFar = number;
+            }
+
+            return number + " is prime.";
+        });
+    }
+
+    @GET
+    @Path("/prime/highest")
+    public String highestPrimeNumberSoFar() {
+        return "The highest prime number so far is " + highestPrimeNumberSoFar + ".";
+    }
+
+    @GET
+    @Path("/duplicates")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String duplicates() {
+        originalCounter.increment();
+        return "duplicated metrics";
+    }
+
+    @GET
+    @Path("/duplicates2")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String duplicates2() {
+        duplicatedCounter.increment();
+        return "duplicated metrics";
+    }
```

Comment:
Please add a method for root page, always helpful for users not following instructions, e.g. adapted from OpenAPI quickstart:

@GET
    @Produces(MediaType.TEXT_PLAIN)
    public String getRootResponse() {
        return "MicroProfile Micrometer quickstart deployed successfully. You can find the available operations in the included README file.";
    }  

PS: if you agree and add this then on BasicRuntimeTestIT please check this endpoint instead

Reply:
Done.